catalog/core/pom.xml

```diff
@@ -104,5 +104,9 @@
         <module>catalog-core-metacardtype</module>
         <module>catalog-core-attachment</module>
         <module>catalog-core-attachment-impl</module>
+        <module>catalog-core-provider</module>
+        <module>catalog-core-solr-api</module>
+        <module>catalog-core-solr-impl</module>
+        <module>catalog-core-solr-defaultsolrcollectionprovider</module>
```

Comment:
❓ These appear to be all new components. What's keeping them from living outside of this repo and contributed back later?

Reply:
that's a possibility. @pklinef is pending a larger effort to take the solr component out of DDF in its entirety as a separate repo. I want to introduce this PR to the baseline first and have it taken out all together at once later.